### PR TITLE
Fix fastify routes command

### PR DESCRIPTION
**Pull request by lundibundi**
Fixes: https://github.com/fastify/fastify-cli/issues/258

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

I've added a test though it may be flaky so I'm not sure we want it.

### CODE REVIEW

**Review by fox1t — APPROVED** [review]
LGTM

File: test/print-routes.test.js [hunk 1]
```diff
@@ -22,6 +22,21 @@ test('should print routes', async t => {
...
 })
 
+test('should print routes via cli', t => {
+  t.plan(2)
+
+  const spy = sinon.spy()
+  const command = proxyquire('../print-routes', {
+    './log': spy
+  })
+  command.cli(['./examples/plugin.js'])
```
Comment (Eomm) [inline test/print-routes.test.js:32]: Could you add a test with the `--help`?

Reading the code, this case should not return a promise, so the `.then` should break.
So maybe it is necessary to move the `if(fastify)` in the upperscope

Comment (lundibundi) [reply test/print-routes.test.js:32]: I'll look in detail later but from what I've seen the `help` case uses explicit `process.exit` so that `.then` will never execute.

Comment (Eomm) [reply test/print-routes.test.js:32]: You are right!

https://github.com/fastify/fastify-cli/blob/056c0b510b27d886ddb9c68be63fca42fc37b935/util.js#L12

(in this case, the `if` statement should not be necessary)

File: test/print-routes.test.js [hunk 2]
```diff
@@ -22,6 +22,21 @@ test('should print routes', async t => {
...
+
+  setTimeout(() => {
+    t.ok(spy.called)
+    t.ok(spy.calledWithMatch('debug', '└── / (GET|POST)\n'))
+  }, 5000)
```
Comment (mcollina) [inline test/print-routes.test.js:37]: why is this needed? can you please use fake timers? Or maybe refactor to avoid needing this in the first place?

Comment (lundibundi) [reply test/print-routes.test.js:37]: The `cli` doesn't return a promise or any other indication that it has completely finished so I opted in to just wait a bit.
I'm not sure what would be the best approach here. I could make the `cli` return a promise but that's imo not a very good solution. We could also use `process on exit` though I'm not sure how `tap` works with that. I'm not sure how I can use fake timers to handle that though.

Also, as I mentioned in a PR comment we could just avoid this test.

Comment (mcollina) [reply test/print-routes.test.js:37]: Let's make the cli return a promise instead.

Comment (lundibundi) [reply test/print-routes.test.js]: Done

**Comment by lundibundi** [comment]
ping @fox1t @mcollina @Ethan-Arrowood I think this can be landed. =)
